pandas/core/generic.py

```diff
@@ -6640,7 +6640,11 @@ def replace(
 
                 for k, v in items:
                     keys, values = list(zip(*v.items())) or ([], [])
-                    if set(keys) & set(values):
+                    # add another check to avoid boolean being regarded
+                    # as binary in python set
+                    if set(keys) & set(values) and set(map(str, keys)) & set(
+                        map(str, values)
```

Comment:
I think this might be a little bit too permissive now, as it will allow `{0: 1.0, 1: 'a'}`, which was previously rejected (might not actually matter but is a change in behavior we should be cognizant of).

Reply:
yeah, i should have thought of it more thoroughly

Reply:
Can this just not be removed altogether? Not clear on the purpose of it